File: client/src/services/ttsService.ts
import { apiClient } from './apiClient'

export interface TtsRequest {
  text: string
  voiceName?: string
  languageCode?: string
  audioEncoding?: string
}

export interface TtsResponse {
  audioContent: string
  audioUrl?: string
}

export interface Voice {
  name: string
  languageCode: string
  displayName: string
}

class TtsService {
  private baseUrl = '/api/tts'

  async generateAudio(request: TtsRequest): Promise<TtsResponse> {
    const response = await apiClient.post(`${this.baseUrl}/generate`, request)
    return response.data
  }

  async synthesizeAudio(request: TtsRequest): Promise<TtsResponse> {
    const response = await apiClient.post(`${this.baseUrl}/synthesize`, request)
    return response.data
  }

  async getAvailableVoices(): Promise<Voice[]> {
    const response = await apiClient.get(`${this.baseUrl}/voices`)
    return response.data
  }

  async checkHealth(): Promise<boolean> {
    try {
      const response = await apiClient.get(`${this.baseUrl}/health`)
      return response.status === 200
    } catch {
      return false
    }
  }
}

export const ttsService = new TtsService() 

File: server/tts-service/src/main/java/com/flexfit/ttsservice/service/TtsService.java
package com.flexfit.ttsservice.service;

import com.flexfit.ttsservice.dto.TtsRequest;
import com.flexfit.ttsservice.dto.TtsResponse;
import org.springframework.core.io.Resource;

public interface TtsService {
    
    /**
     * Convert text to speech and return audio as byte array
     * @param request TTS request containing text and voice parameters
     * @return Audio data as byte array
     */
    byte[] textToSpeech(TtsRequest request);
    
    /**
     * Convert text to speech and return full response with metadata
     * @param request TTS request containing text and voice parameters
     * @return TTS response with audio URL and metadata
     */
    TtsResponse generateAudio(TtsRequest request);
    
    /**
     * Get available voices from Google Cloud TTS
     * @return List of available voices
     */
    String getAvailableVoices();
} 